src/main/scala/com/fulcrumgenomics/umi/Umis.scala

```diff
@@ -80,8 +80,9 @@ object Umis {
       Some(name.substring(idx + 1, name.length))
     }
 
-    val umi   = rawUmi.map(raw => (if (raw.indexOf('+') > 0) raw.replace('+', '-') else raw).toUpperCase)
-    val valid = umi.forall(u => u.forall(isValidUmiCharacter))
+    val umiSeq = rawUmi.map(seq => (if (prefix.isEmpty) seq else seq.stripPrefix(prefix.get)))
+    val umi = umiSeq.map(raw => (if (raw.indexOf('+') > 0) raw.replace('+', '-') else raw).toUpperCase)
+    val valid  = umi.forall(u => u.forall(isValidUmiCharacter))
 
```

Comment:
We also tend to align equals in blocks in the codebase (I have an IntelliJ plugin to do this):

```suggestion
    val umiSeq = rawUmi.flatMap(raw => prefix.map(raw.stripPrefix))
    val umi    = umiSeq.map(raw => (if (raw.indexOf('+') > 0) raw.replace('+', '-') else raw).toUpperCase)
    val valid  = umi.forall(u => u.forall(isValidUmiCharacter))

```